### PR TITLE
Add `effective_name` property to `Chat` object

### LINKED ISSUE #3480: [FEATURE] convenience property for displaying name of Chat object
### What kind of feature are you missing? Where do you notice a shortcoming of PTB?

Because `Chat` objects can be private or groups/user/channels, one needs to check the type every time before using the appropriate property (`Chat.full_name` or `Chat.title` respectively). It would be nice to have a conveniente property that does this automatically

### Describe the solution you'd like

I propose a convenience property that return `Chat.full_name` if the chat.type is private or `Chat.title` otherwise.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Viicos**
Fixes #3480.

### Checklist for PRs

- [x] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing

### CODE REVIEW

**Comment by Viicos** [comment]
Let me know which `.. versionadded::` should be used, and if this property name is fine

**Comment by Poolitzer** [comment]
@Viicos I would call it <code>effective_name</code>, and it needs a test

**Comment by clot27** [comment]
> Let me know which `.. versionadded::` should be used, and if this property name is fine

the next release will be v20.1

File: telegram/_chat.py [hunk 1]
```diff
@@ -388,10 +388,24 @@ def __init__(
...
+        :obj:`str`: Convenience property. Gives :attr:`title` if not :obj:`None`,
+        else :attr:`full_name` if not :obj:`None`.
+
+        .. versionadded:: 20.1
+        """
+        if self.title is not None:
+            return self.title
+        if self.full_name is not None:
+            return self.full_name
+        return None
```
Comment (clot27) [inline telegram/_chat.py:403]: You are not testing this line, please add on as:
```
chat = Chat(id=1, type=Chat.GROUP)
assert chat.effective_name is None

```

Comment (Viicos) [reply telegram/_chat.py:403]: Yes saw that, I'm on something else right now but this will be done in ~30mins

**Review by clot27 — APPROVED** [review]
LGTM

**Comment by Bibo-Joshi** [comment]
Thank you very much for your contribution :)

**Pull request merged**